includes/class-freemius.php

```diff
@@ -4034,42 +4034,49 @@ private function should_run_connectivity_test( $flush_if_no_connectivity = false
         }
 
         /**
-         * @author Vova Feldman (@svovaf)
-         * @since  1.1.7.4
+         * @author Leo Fajardo (@leorw)
+         * @since  2.5.3
          *
-         * @param int|null $blog_id      Since 2.0.0.
-         * @param bool     $is_gdpr_test Since 2.0.2. Perform only the GDPR test.
+         * @param bool $is_update
          *
-         * @return object|false
+         * @return bool
          */
-        private function ping( $blog_id = null, $is_gdpr_test = false ) {
-            if ( WP_FS__SIMULATE_NO_API_CONNECTIVITY ) {
-                return false;
+        private function should_turn_fs_on( $is_update = true ) {
+            if ( empty( $this->_plugin->moderation ) ) {
+                return true;
             }
 
-            $version = $this->get_plugin_version();
+            $optin_config = $this->_plugin->moderation['opt_in'];
 
-            $is_update = $this->apply_filters( 'is_plugin_update', $this->is_plugin_update() );
+            if (
+                WP_FS__IS_LOCALHOST_FOR_SERVER &&
+                ( empty( $optin_config['localhost'] ) || true === $optin_config['localhost'] )
+            ) {
+                return true;
+            }
 
-            $params = array(
-                'is_update'    => json_encode( $is_update ),
-                'version'      => $version,
-                'sdk'          => $this->version,
-                'is_admin'     => json_encode( is_admin() ),
-                'is_ajax'      => json_encode( self::is_ajax() ),
-                'is_cron'      => json_encode( self::is_cron() ),
-                'is_gdpr_test' => $is_gdpr_test,
-                'is_http'      => json_encode( WP_FS__IS_HTTP_REQUEST ),
-            );
+            $visibility = $is_update ?
+                $optin_config['updates'] :

```

Comment:
@fajardoleo similar to the comment above, in this case, we need to validate the array keys exist with `isset()` before accessing.